tests/test_photon_ops.py

```diff
@@ -60,3 +63,74 @@ def test_lsst_optics() -> None:
         world_origin=galsim.PositionD(x=0.0, y=0.0),
     )
     lsst_optics.applyTo(photon_array, local_wcs=local_wcs)
+
+
+def test_lsst_diffraction() -> None:
+    """This just makes sure that the LsstDiffractionFactory runs.
+    It does not check plausibility of results."""
+    boresight = galsim.CelestialCoord(0.543 * galsim.radians, -0.174 * galsim.radians)
+
+    det_name = "R22_S11"
+    lsst_diffraction = photon_ops.LsstDiffraction(
+        telescope=load_telescope(telescope="LSST", band="r"),
+        sky_pos=galsim.CelestialCoord(0.543 * galsim.radians, -0.174 * galsim.radians),
+        icrf_to_field=create_test_icrf_to_field(boresight, det_name),
+        latitude=-30.24463,
+        azimuth=45.0,
+        altitude=45.0,
+        telescope_rotation_angle=0.0,
+    )
+    photon_array = galsim.PhotonArray(
+        5,
+        x=np.array([-0.04823635, 0.47023422, -8.53736263, 0.8639109, -3.0237201]),
+        y=np.array([1.76626949, -0.89284146, 13.51962823, 0.82503544, -0.1011734]),
+        wavelength=np.array(
+            [577.67626034, 665.6715595, 564.75533946, 598.74363606, 571.04519139]
+        ),
+        flux=np.ones(5),
+        pupil_u=np.array(
+            [-3.60035156, -2.25328125, -2.31042969, 2.56351562, -0.46535156]
+        ),
+        pupil_v=np.array(
+            [1.00417969, 2.73496094, -2.92273437, -1.16746094, 3.09417969]
+        ),
+    )
+    local_wcs = galsim.AffineTransform(
+        0.168,
+        0.108,
+        -0.108,
+        0.168,
+        origin=galsim.PositionD(x=-0.349, y=-0.352),
+        world_origin=galsim.PositionD(x=0.0, y=0.0),
+    )
+    lsst_diffraction.applyTo(photon_array, local_wcs=local_wcs)
+
+
+def test_xy_to_v():
+    """Tests if the transform photon_ops.XyToV and its inverse combine to
+    the identity operation."""
+    galsim.CelestialCoord(0.543 * galsim.radians, -0.174 * galsim.radians)
+    local_wcs = galsim.AffineTransform(
+        0.168,
+        0.108,
+        -0.108,
+        0.168,
+        origin=galsim.PositionD(x=-0.349, y=-0.352),
+        world_origin=galsim.PositionD(x=0.0, y=0.0),
+    )
+    icrf_to_field = create_test_icrf_to_field(
+        boresight=galsim.CelestialCoord(
+            0.543 * galsim.radians, -0.174 * galsim.radians
+        ),
+        det_name="R22_S11",
+    )
+    sky_pos = galsim.CelestialCoord(0.543 * galsim.radians, -0.174 * galsim.radians)
+    xy_to_v = photon_ops.XyToV(local_wcs, icrf_to_field, sky_pos)
+
+    x, y = np.array(
+        np.meshgrid(np.linspace(-10.0, 10, 20), np.linspace(-10.0, 10, 20))
+    ).reshape((2, -1))
+    v = xy_to_v(x, y)
+    x_after, y_after = xy_to_v.inverse(v)
+    np.testing.assert_array_almost_equal(x, x_after)
+    np.testing.assert_array_almost_equal(y, y_after)
```

Comment:
Indeed, the math is pretty involved. To be honest it is not transparent at all, what is happening behind all the WCS's involved here. It would take me quite a while to fully understand the transforms.
Of course, any help is welcome here, but I would not be able to provide a good test until the end of the week.

At least, I tried mapping `(x,y)=(0,0)` to `v`, but the result `v=(3.40953136e-13, -6.53235082e-04, -9.99999787e-01)` also does not look like something predictable.

Reply:
I was thinking of some special cases like when the sky_pos is close to boresight, then the projection stuff is nearly trivial.  So (x,y) near (0,0) should be straightforward.  Even moreso if the wcs is just a PixelScale, but maybe even simple in the general local_wcs case.

Reply:
I just noticed, that in the current example `sky_pos == boresight`.
But it is still not transparent to me what exactly is happening behind each of the steps involved in this transform